libphal/phal_sbe.C

```diff
@@ -55,6 +52,61 @@ void validateSBEState(struct pdbg_target *proc)
 		throw sbeError_t(exception::SBE_CHIPOP_NOT_ALLOWED);
 	}
 }
+void setState(struct pdbg_target *proc, enum sbe_state state)
+{
+	// get PIB target
+	struct pdbg_target *pib = getPibTarget(proc);
+	if (sbe_set_state(pib, state)) {
+		log(level::ERROR,
+		    "Failed to set SBE state(%d) information (%s)", state,
+		    pdbg_target_path(proc));
+		throw sbeError_t(exception::SBE_STATE_WRITE_FAIL);
+	}
+}
+
+enum sbe_state getState(struct pdbg_target *proc)
+{
+	// get PIB target
+	struct pdbg_target *pib = getPibTarget(proc);
+	enum sbe_state state;
+
+	if (sbe_get_state(pib, &state)) {
+		log(level::ERROR, "Failed to get SBE state information (%s)",
+		    pdbg_target_path(proc));
+		throw sbeError_t(exception::SBE_STATE_READ_FAIL);
+	}
+	return state;
+}
+
+bool isIPLDone()
+{
+	// get primary processor target
+	struct pdbg_target *proc = getPrimaryProc();
+
+	// get pib target associated to primary processor
+	struct pdbg_target *pib = getPibTarget(proc);
```

Comment:
Need to re-throw exception if getPrimaryTarget() or getPibTarget() throws an exception?